debuginfo/src/dwarf.rs

```diff
@@ -399,13 +399,6 @@ impl<'d, 'a> DwarfUnit<'d, 'a> {
             None => return Err(gimli::read::Error::MissingUnitDie.into()),
         };
 
-        // Clang's LLD might eliminate an entire compilation unit and simply set the low_pc to zero
-        // and remove all range entries to indicate that it is missing. Skip such a unit, as it does
-        // not contain any code that can be executed.
-        if unit.low_pc == 0 && entry.attr(constants::DW_AT_ranges)?.is_none() {
-            return Ok(None);
```

Comment:
This change might have negative performance impact on our end. Could you explain the reason why this is being removed? I'm curious if we missed some case here.

Reply:
As the commit message says, zero addresses occur in practice in `.o` files, and without this change the debug info for `.o` files is completely skipped.

If it is a performance problem, I wonder if the code could detect that it's a `.o`, e.g. by recognizing that there is a single nameless section, setting a flag accordingly, and then incorporating the flag into this test?

Reply:
@jan-auer: Any thoughts on my suggestion?